pandas/core/generic.py

```diff
@@ -8763,6 +8763,51 @@ def transform(self, func, *args, **kwargs):
         scalar : type of index
         """
 
+    def _find_first_valid(self, reversed=False):
+        """Retrieves the index of the first valid value.
+
+        Parameters
+        ----------
+        reversed : bool
+            If you want invert the original data order (axis=0). Useful to find
+            the last valid value.
+
+        Returns
+        -------
+        idx_first_valid : type of index
+        """
+        if len(self) == 0:  # early stop
+            return None
+        is_valid = ~self.isna()
+
+        if self.ndim == 2:
+            is_valid = is_valid.any(1)  # reduce axis 1
+
+        if not reversed:
+            # first valid value case
+            i = is_valid.idxmax()
+            if not is_valid[i]:
+                return None
+            else:
```

Comment:
don't need the elses (just return)

Reply:
hmmmm, I'm not sure, I was thinking in the case when there is no valid values (all nans), In that case this function returns the first index when should be `None`.